src/app/components/import/ImportCsvModalContent.js

```diff
@@ -0,0 +1,85 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { c } from 'ttag';
+import { Table, Alert, Block } from 'react-components';
+
+import ImportCsvTableHeader from './ImportCsvTableHeader';
+import ImportCsvTableBody from './ImportCsvTableBody';
+
+import { prepare, toVcardContacts } from '../../helpers/csv';
+import { modifyContactField, modifyContactType, toggleContactChecked } from '../../helpers/import';
+
+const ImportCsvModalContent = ({ file, onSetVcardContacts }) => {
+    const [isParsingFile, setIsParsingFile] = useState(true);
+    const [contactIndex, setContactIndex] = useState(0);
+    const [preVcardsContacts, setpreVcardsContacts] = useState([]);
+
+    const handleClickPrevious = () => setContactIndex((index) => index - 1);
+    const handleClickNext = () => setContactIndex((index) => index + 1);
+
+    const handleToggle = (groupIndex) => (index) =>
+        setpreVcardsContacts(preVcardsContacts.map((contact) => toggleContactChecked(contact, [groupIndex, index])));
+
+    const handleChangeField = (groupIndex) => (newField) =>
+        setpreVcardsContacts(preVcardsContacts.map((contact) => modifyContactField(contact, groupIndex, newField)));
+
+    const handleChangeType = (groupIndex) => (newType) =>
+        setpreVcardsContacts(preVcardsContacts.map((contact) => modifyContactType(contact, groupIndex, newType)));
+
+    useEffect(() => {
+        const parseFile = async () => {
```

Comment:
Move `parseFile` outside of the `useEffect` callback please.

Reply:
Do you mean moving the definition of `parseFile` outside the `useEffect` hook? Or rather remove the `useEffect` hook completely?
It's true I can do something like:
```
const parseFile = async () => {
    const preVcardsContacts = prepare(file);
    setpreVcardsContacts(preVcardsContacts);
    setIsFileParsed(true);
};
if (!isFileParsed) {
    parseFile();
}
```
and get rid of that `useEffect` hook. But I thought it was more clear that this function should only be run once when the component is loaded by using the structure `useEffect(() => ... , [])`